src/languages/lib/swift.js

```diff
@@ -0,0 +1,285 @@
+import { concat, either } from '../../lib/regex';
+
+// Keywords that require a leading dot.
+export const dotKeywords = [
+  /Protocol\b/, // contextual
+  /Type\b/, // contextual
+];
+
+// Keywords that may have a leading dot.
+export const optionalDotKeywords = [
+  /init\b/,
+  /self\b/
+];
+
+// Regular keywords and literals.
+export const keywords = [
+  /\bAny\b/,
+  /\bSelf\b/,
+  /\b_\b/, // pattern
+  /\bassociatedtype\b/,
+  /\bas\?\B/, // operator
+  /\bas!\B/, // operator
+  /\bas\b/, // operator
+  /\bbreak\b/,
+  /\bcase\b/,
+  /\bcatch\b/,
+  /\bclass\b/,
+  /\bcontinue\b/,
+  /\bconvenience\b/, // contextual
+  /\bdefault\b/,
+  /\bdefer\b/,
+  /\bdeinit\b/,
+  /\bdidSet\b/, // contextual
+  /\bdo\b/,
+  /\bdynamic\b/, // contextual
+  /\belse\b/,
+  /\benum\b/,
+  /\bextension\b/,
+  /\bfallthrough\b/,
+  /\bfalse\b/, // literal
+  /\bfileprivate\(set\)\B/,
+  /\bfileprivate\b/,
+  /\bfinal\b/, // contextual
+  /\bfor\b/,
+  /\bfunc\b/,
+  /\bget\b/, // contextual
+  /\bguard\b/,
+  /\bif\b/,
+  /\bimport\b/,
+  /\bindirect\b/, // contextual
+  /\binfix\b/, // contextual
+  /\binit\?\B/,
+  /\binit!\B/,
+  /\binout\b/,
+  /\binternal\(set\)\B/,
+  /\binternal\b/,
+  /\bin\b/,
+  /\bis\b/, // operator
+  /\blazy\b/, // contextual
+  /\blet\b/,
+  /\bmutating\b/, // contextual
+  /\bnil\b/, // literal
+  /\bnonmutating\b/, // contextual
+  /\bopen\(set\)\B/, // contextual
+  /\bopen\b/, // contextual
+  /\boperator\b/,
+  /\boptional\b/, // contextual
+  /\boverride\b/, // contextual
+  /\bpostfix\b/, // contextual
+  /\bprecedencegroup\b/,
+  /\bprefix\b/, // contextual
+  /\bprivate\(set\)\B/,
+  /\bprivate\b/,
+  /\bprotocol\b/,
+  /\bpublic\(set\)\B/,
+  /\bpublic\b/,
+  /\brepeat\b/,
+  /\brequired\b/, // contextual
+  /\brethrows\b/,
+  /\breturn\b/,
+  /\bset\b/, // contextual
+  /\bsome\b/, // contextual
+  /\bstatic\b/,
+  /\bstruct\b/,
+  /\bsubscript\b/,
+  /\bsuper\b/,
+  /\bswitch\b/,
+  /\bthrows\b/,
+  /\bthrow\b/,
+  /\btrue\b/, // literal
+  /\btry\?\B/, // operator
+  /\btry!\B/, // operator
+  /\btry\b/, // operator
+  /\btypealias\b/,
+  /\bunowned\(safe\)\B/, // contextual
+  /\bunowned\(unsafe\)\B/, // contextual
+  /\bunowned\b/, // contextual
+  /\bvar\b/,
+  /\bweak\b/, // contextual
+  /\bwhere\b/,
+  /\bwhile\b/,
+  /\bwillSet\b/, // contextual
+];
+
+// NOTE: Contextual keywords are reserved only in specific contexts.
+// Ideally, these should be matched using modes to avoid false positives.
+
+// TODO: Create a PRECEDENCE_GROUP mode to match the remaining contextual keywords:
+// assignment associativity higherThan left lowerThan none right
+// These aren't included in the list because they result in mostly false positives.
+
+// Keywords that start with a number character (#).
+// #available is handled separately.
+export const numberKeywords = [
```

Comment:
```suggestion
export const numberSignKeywords = [
```

Otherwise it's just ambiguous and confusing.